### PR TITLE
Add initial implementation of turso integration

### PR DESCRIPTION
**Pull request by rylev**
Marking this as WIP since it needs a bit of clean up, but it works in its current state. I was able to run the [example todo app](https://github.com/rylev/spin-todo) against a Turso backend.

### CODE REVIEW

File: crates/trigger/src/runtime_config/sqlite.rs [hunk 1]
```diff
@@ -53,6 +52,7 @@ fn execute_statements(
 #[serde(rename_all = "snake_case", tag = "type")]
 pub enum SqliteDatabaseOpts {
     Spin(SpinSqliteDatabaseOpts),
+    Turso(TursoOpts),
```
Comment (radu-matei) [inline crates/trigger/src/runtime_config/sqlite.rs:55]: Thoughts vs. naming this a `libsql` backend?

File: crates/sqlite-turso/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+fn convert_parameters(parameters: &[spin_world::sqlite::Value]) -> Vec<libsql_client::Value> {
+    parameters
+        .iter()
+        .map(|v| match v {
+            sqlite::Value::Integer(value) => libsql_client::Value::Integer { value: *value },
+            sqlite::Value::Real(value) => libsql_client::Value::Float { value: *value },
+            sqlite::Value::Text(t) => libsql_client::Value::Text { value: t.clone() },
+            sqlite::Value::Blob(_) => todo!(),
```
Comment (radu-matei) [inline crates/sqlite-turso/src/lib.rs:75]: Does LibSQL support blob values?

Comment (radu-matei) [inline crates/sqlite-turso/src/lib.rs:75]:
```suggestion
            sqlite::Value::Blob(b) => libsql_client::Value::Blob { value: b.clone() },
```

File: crates/trigger/src/runtime_config/sqlite.rs (hunk 1 above)
Comment (rylev) [reply crates/trigger/src/runtime_config/sqlite.rs:55]: I don't think it's really about libsql as that can be run locally and should work fine with the default spin configuration. Perhaps `sqld` or `remote_libsql`?

Comment (radu-matei) [reply crates/trigger/src/runtime_config/sqlite.rs:55]: The library is `libsql-client`. LibSQL is intended to support both local scenarios and remote scenarios, depending on the URI, so I feel like calling it `libsql` wouldn't be inappropriate.

Comment (rylev) [reply crates/trigger/src/runtime_config/sqlite.rs:55]: Should we also allow running libsql locally as well?

**Comment by itowlson** [comment]
Incorporated into #1545.

**Pull request closed without merging**